### PR TITLE
Issue/block tracker may panic

### PR DESCRIPTION
**Pull request by ronnno**
Fixed a couple of race conditions. 
see #95 
Did not provide tests for race conditions.

### CODE REVIEW

File: services/statestorage/block_tracker.go [hunk 1]
```diff
@@ -29,32 +30,45 @@ func NewBlockTracker(startingHeight uint64, graceDist uint16, timeout time.Durat
...
+	defer t.mutex.RUnlock()
+
+	return t.currentHeight, t.latch
+}
+
 func (t *BlockTracker) WaitForBlock(requestedHeight primitives.BlockHeight) error {
 
 	rh := uint64(requestedHeight)
-	if t.currentHeight >= rh { // requested block already committed
+	ch, cl := t.readAtomicHeightAndLatch()
```
Comment (electricmonk) [inline services/statestorage/block_tracker.go]: please use more verbose names

File: services/statestorage/block_tracker.go [hunk 2]
```diff
@@ -29,32 +30,45 @@ func NewBlockTracker(startingHeight uint64, graceDist uint16, timeout time.Durat
...
-	if t.currentHeight < rh-t.graceDistance { // requested block too far ahead, no grace
+	if ch < rh-t.graceDistance { // requested block too far ahead, no grace
 		return errors.Errorf("requested future block outside of grace range")
 	}
 
 	timer := time.NewTimer(t.timeout)
 	defer timer.Stop()
 
-	for t.currentHeight < rh { // sit on latch until desired height or t.o.
+	for ch < rh { // sit on latch until desired height or t.o.
```
Comment (electricmonk) [inline services/statestorage/block_tracker.go]: if the variable names were more verbose you wouldn't need the comment here

File: services/statestorage/block_tracker.go (hunk 1 above)
Comment (ronnno) [reply services/statestorage/block_tracker.go]: done

File: services/statestorage/block_tracker.go (hunk 2 above)
Comment (ronnno) [reply services/statestorage/block_tracker.go]: removed comment

**Pull request merged**